backend/routes/recipeRouter.js

```diff
@@ -34,11 +34,46 @@ router.get( '/:id', (req, res) => {
 
 
 /* PUT */
-
+router.put('/:id', (req, res) => {
+  const { id } = req.params;
+  const recipeEdit = req.body;
+
+  if (recipeEdit.name && recipeEdit.link) {
+    recipes.update(id, recipeEdit)
+      .then((recipe) => {
+        if (id) {
+          res.json({ message: "Recipe has been updated." })
+        } else {
+          res.status(400).json({ message: "Recipe with specified ID does not exist." })
+        }
+      })
+      .catch((err) => {
+        res.status(500).json({ message: "Failed to update recipe." })
+      })
+  } else {
+    res.status(404).json({
+      message: "Missing name or link."
+    })
+  }
+});
 
 
 /* DELETE */
+router.delete('/:id', (req, res) => {
+  const { id } = req.params;
 
+  recipes.remove(id)
+    .then((recipe) => {
+      if (id) {
+        res.json({ message: "User has been deleted." })
+      } else {
+        res.status(400).json({ message: "Recipe with specified ID does not exist." })
```

Comment:
```suggestion
        res.status(400).json({ message: "ID not provided." })
```

If the recipe didn't exist, it would be a 404 message